src/controllers/controllermanager.cpp

```diff
@@ -280,16 +286,12 @@ void ControllerManager::stopPolling() {
 
 void ControllerManager::pollDevices() {
     Trace tracer("ControllerManager::pollDevices");
-    bool eventsProcessed(false);
     // Continue to poll while any device returned data.
-    do {
-        eventsProcessed = false;
-        foreach (Controller* pDevice, m_controllers) {
-            if (pDevice->isOpen() && pDevice->isPolling()) {
-                eventsProcessed = pDevice->poll() || eventsProcessed;
-            }
+    foreach (Controller* pDevice, m_controllers) {
+        if (pDevice->isOpen() && pDevice->isPolling()) {
+            pDevice->poll();
         }
-    } while (eventsProcessed);
```

Comment:
Has this been tested with chatty controllers, such as those with high-resolution wheels (1000 ticks per revolution or so?) The reason we included a while loop was to ensure all pending messages were processed ASAP.